source/game/system/map/MapdataCheckPoint.hh

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include "game/system/map/MapdataAccessorBase.hh"
+
+#include <egg/math/Vector.hh>
+
+namespace System {
+
+class MapdataCheckPoint;
+class MapdataCheckPointAccessor;
+
+struct LinkedCheckpoint {
+    MapdataCheckPoint *checkpoint;
+    EGG::Vector2f p0diff;
+    EGG::Vector2f p1diff;
+    f32 distance;
+};
+
+class MapdataCheckPoint {
+public:
+    static constexpr s8 NORMAL_CHECKPOINT = -1; ///< only used for picking respawn position
+    static constexpr s8 FINISH_LINE = 0;        ///< triggers a lap change; also the starting line
```

Comment:
To match formatting across other files, since these are static, they should go after all non-static variable declarations. Since these are only ever used by `MapdataCheckPoint`, they should ALSO be `private`. So move them after `LinkedCheckpoint m_nextPoints[6];`

Reply:
they're actually also used by `MapdataCheckPointAccessor` so they can't be private: 
```cpp
void MapdataCheckPointAccessor::findFinishAndLastKcp() {
        //...
        if (checkpoint->type() == MapdataCheckPoint::FINISH_LINE) {
            finishLineCheckpointId = ckptId;
        }
        //...
}
```

unless we resurrect the `isFinishLine` getter

Reply:
I specifically did not suggest removing `isFinishLine` because of this very reason :P You'll need to add it back.